firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift

```diff
@@ -98,9 +145,23 @@ class FormAutofillHelper: TabContentScript {
         ccPlainText.ccNumber = "\(payload.ccNumber.filter { $0.isNumber })"
         ccPlainText.ccNumberLast4 = ccPlainText.ccNumber.count > 4 ? String(ccPlainText.ccNumber.suffix(4)) : ""
         ccPlainText.ccType = creditCardValidator.cardTypeFor(ccPlainText.ccNumber)?.rawValue ?? ""
-        return ccPlainText
+        return AutofillFieldValuePayload(fieldValue: .creditCard, fieldData: ccPlainText)
     }
 
+    func getFieldTypeValues(payload: AddressAutofillPayload) -> AutofillFieldValuePayload {
+        var addressPlainText = UnencryptedAddressFields()
+
+        addressPlainText.addressLevel1 = payload.addressLevel1
+        addressPlainText.organization = payload.organization
+        addressPlainText.country = payload.country
+        addressPlainText.addressLevel2 = payload.addressLevel2
+        addressPlainText.email = payload.email
+        addressPlainText.streetAddress = payload.streetAddress
+        addressPlainText.name = payload.name
+        addressPlainText.postalCode = payload.postalCode
```

Comment:
nit: you already have the init so curious if there is a reason we don't directly set these values? 🤔 
```suggestion
        var addressPlainText = UnencryptedAddressFields(
            addressLevel1: payload.addressLevel1,
            organization: payload.organization,
            country: payload.country,
            addressLevel2: payload.addressLevel2,
            email: payload.email,
            streetAddress: payload.streetAddress,
            name: payload.name,
            postalCode: payload.postalCode
        )
```